modules/apps/image-uploader/image-uploader-web/src/main/java/com/liferay/image/uploader/web/internal/upgrade/v1_0_1/DLFolderNameUpgradeProcess.java

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright (c) 2000-present Liferay, Inc. All rights reserved.
+ *
+ * This library is free software; you can redistribute it and/or modify it under
+ * the terms of the GNU Lesser General Public License as published by the Free
+ * Software Foundation; either version 2.1 of the License, or (at your option)
+ * any later version.
+ *
+ * This library is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.image.uploader.web.internal.upgrade.v1_0_1;
+
+import com.liferay.document.library.kernel.model.DLFolder;
+import com.liferay.document.library.kernel.service.DLFolderLocalService;
+import com.liferay.image.uploader.web.internal.util.UploadImageUtil;
+import com.liferay.petra.string.StringBundler;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Lily Chi
+ */
+public class DLFolderNameUpgradeProcess extends UpgradeProcess {
+
+	public DLFolderNameUpgradeProcess(
+		DLFolderLocalService dlFolderLocalService) {
+
+		_dlFolderLocalService = dlFolderLocalService;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement preparedStatement = connection.prepareStatement(
+				StringBundler.concat(
+					"select DLFolder.folderId from DLFolder inner join ",
```

Comment:
@slnn @yuhai Can you briefly explain what kind of DLFolder we want to check in the commit message, so that anyone looking at this SQL can get a better understanding?

Reply:
Hi @dantewang 

From my debug process, I found this, the target DLFolder should satisfy these:

1.  The value of DLFolder.name should be `java.lang.Class`
2.  The target DLFolder's parent folder's name should be the current user's userId
3. The target DLFolder's parent folder's parent folder's name should be the `_UUID` in TempFileEntryUtil
4. The name of the root DLFolder in the tree must be `com.liferay.portal.kernel.util.TempFileEntryUtil`, and it must be a hidden DLFolder which means DLFolder.hidden_ must be 1.


Lily

Reply:
@slnn, please you modify the [commit ](https://github.com/liferay-core-infra/liferay-portal/pull/4664/commits/b2c75131f54dd133095f0a7f39e74bbb1ed57778) message  for 

When uploading an image for user profile, it will separately create every level directory as DLfolder data. For example, "temp\00ef1134-b3ee-432a-babd-367cefa44de1\20123\java.lang.Class" , for the "java.lang.Class", the fix changed its name, so need to use upgrade process to update the old folderName.

Reply:
Force pushed with the update for the commit message, no logic changes

Reply:
@dantewang, we just changed the commit message, is it good?